h2o-py/h2o/explanation/_explain.py

```diff
@@ -1752,11 +1752,312 @@ def residual_analysis_plot(
     return fig
 
 
+def learning_curve_plot(
+        model,  # type: h2o.model.ModelBase
+        metric="AUTO",  # type: str
+        cv_ribbon=None,  # type: Optional[bool]
+        cv_lines=None,  # type: Optional[bool]
+        figsize=(16,9),  # type: Tuple[float]
+        colormap=None  # type: Optional[str]
+):
+    # type: (...) -> plt.Figure
+    """
+    Learning curve
+
+    Create learning curve plot for an H2O Model. Learning curves show error metric dependence on
+    learning progress, e.g., RMSE vs number of trees trained so far in GBM. There can be up to 4 curves
+    showing Training, Validation, Training on CV Models, and Cross-validation error.
+
+    :param model: an H2O model
+    :param metric: a stopping metric
+    :param cv_ribbon: if True, plot the CV mean as a and CV standard deviation as a ribbon around the mean,
+                      if None, it will attempt to automatically determine if this is suitable visualisation
+    :param cv_lines: if True, plot scoring history for individual CV models, if None, it will attempt to
+                     automatically determine if this is suitable visualisation
+    :param figsize: figure size; passed directly to matplotlib
+    :param colormap: colormap to use
+    :return: a matplotlib figure
+
+    :examples:
+    >>> import h2o
+    >>> from h2o.estimators import H2OGradientBoostingEstimator
+    >>>
+    >>> h2o.init()
+    >>>
+    >>> # Import the wine dataset into H2O:
+    >>> f = "https://h2o-public-test-data.s3.amazonaws.com/smalldata/wine/winequality-redwhite-no-BOM.csv"
+    >>> df = h2o.import_file(f)
+    >>>
+    >>> # Set the response
+    >>> response = "quality"
+    >>>
+    >>> # Split the dataset into a train and test set:
+    >>> train, test = df.split_frame([0.8])
+    >>>
+    >>> # Train a GBM
+    >>> gbm = H2OGradientBoostingEstimator()
+    >>> gbm.train(y=response, training_frame=train)
+    >>>
+    >>> # Create the learning curve plot
+    >>> gbm.learning_curve_plot()
+    """
+    if model.algo == "stackedensemble":
+        model = model.metalearner()
+
+    if model.algo not in ("stackedensemble", "glm", "gam", "glrm", "deeplearning",
```

Comment:
would be nice to generalize it, instead of having this enumerated add "an interface" that the algo would implement

this would let us get rid of the big if that determines `allowed_metrics` and `allowed_timesteps`